tests/unity/CMakeLists.txt

```diff
@@ -0,0 +1,181 @@
+      ${test_file_path} ${output_file}
+      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+      RESULT_VARIABLE op_result
+    )
+
+    if (NOT ${op_result} EQUAL 0)
+        message(FATAL_ERROR "Failed to generate runner ${output_file}")
+    endif()
+
+    file(GLOB app_sources ${UNITY_PRODUCTS_DIR}/runner/*.c)
+    target_sources(app PRIVATE ${app_sources})
+
+    message(STATUS "Generating test runner ${unity_runner_arg3}")
+endfunction()
+
+# Generate cmock for provided header file.
+function(cmock_generate header_path dst_path)
+    get_property(RUBY_EXECUTABLE GLOBAL PROPERTY RUBY_EXECUTABLE)
+    get_property(CMOCK_DIR GLOBAL PROPERTY CMOCK_DIR)
+
+    file(MAKE_DIRECTORY "${dst_path}")
+
+    execute_process(
+      COMMAND ${RUBY_EXECUTABLE}
+      ${CMOCK_DIR}/lib/cmock.rb
+      --mock_prefix=mock_
+      --plugins=return_thru_ptr\;ignore_arg\;ignore\;callback\;array
+      --mock_path=${dst_path}
+      ${header_path}
+      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+      RESULT_VARIABLE op_result
+      OUTPUT_VARIABLE output_result
+    )
+
+    if (NOT ${op_result} EQUAL 0)
+          message(SEND_ERROR "${output_result}")
+          message(FATAL_ERROR "Failed to generate runner ${output_file}")
+      endif()
+endfunction()
+
+# Add --wrap linker option for each function listed in the input file.
+function(cmock_linker_trick func_name_path)
+    file(STRINGS ${func_name_path} contents)
+    set(linker_str "-Wl")
+    foreach(src ${contents})
+        set(linker_str "${linker_str},--wrap=${src}")
+    endforeach()
+    zephyr_link_libraries(${linker_str})
+endfunction()
+
+
+# Handle wrapping functions from mocked file.
+# Function takes header file and generates a file containing list of functions.
+# File is then passed to 'cmock_linker_trick' which adds linker option for each
+# function listed in the file.
+function(cmock_linker_wrap_trick header_file_path)
+    set(flist_file "${header_file_path}.flist")
+
+    execute_process(
+      COMMAND
+      ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/../nrf/scripts/unity/func_name_list.py
+      --input ${header_file_path}
+      --output ${flist_file}
+      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+      RESULT_VARIABLE op_result
+      OUTPUT_VARIABLE output_result
+    )
+
+    if (NOT ${op_result} EQUAL 0)
+        message(SEND_ERROR "${output_result}")
+        message(FATAL_ERROR "Failed to parse header ${header_file_path}")
+    endif()
+    cmock_linker_trick(${flist_file})
+endfunction()
+
+# Function takes original header and prepares two version
+# - version with system calls removed and static inline functions
+#   converted to standard function declarations
+# - version with addtional __wrap_ prefix for all functions that
+#   is used to generate cmock
+function(cmock_headers_prepare in_header out_header wrap_header)
+    execute_process(
+      COMMAND
+      ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/../nrf/scripts/unity/header_prepare.py
```

Comment:
`${NRF_DIR}` ?

Reply:
we had conversation about that with @SebastianBoe here https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/519#discussion_r267239960

I would like to if you tell me how to do it 😄

Reply:
Ahh, correct.
See you place `NRF_DIR` in the cache, which I like.
Does make code bit more flexible.